.github/workflows/ci.yml

```diff
@@ -12,14 +12,50 @@ permissions:
 jobs:
   test:
     runs-on: ubuntu-latest
+
     steps:
       - uses: actions/checkout@v3
+
       - name: Set up Python 3.9
         uses: actions/setup-python@v4
         with:
           python-version: "3.9"
           cache: "pip"
           cache-dependency-path: |
             requirements-dev.txt
-      - run: pip install -r requirements-dev.txt
-      - run: pytest
+
+      - name: Install dependencies
+        run: pip install -r requirements-dev.txt
+
+      - name: Test
+        run: pytest
+
+  docs:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+          cache: "pip"
+          cache-dependency-path: |
+            requirements-docs.txt
+
+      - name: Install dependencies
+        run: pip install -r requirements-docs.txt
+
+      - name: Build docs
+        run: talondoc build ./ docs/_build 
+
+      - name: Deploy docs
+        if: github.event_name == 'push'
+        uses: JamesIves/github-pages-deploy-action@v4.4.1
```

Comment:
Given [GitHub Pages supports deploying straight from GitHub Actions](https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/) rather than pushing build artifacts to a branch, whatever workflow we have here should utilise that.

Reply:
Would you be happy to push that change to the PR?

Reply:
I don't have write access to this PR.

Reply:
It seems only maintainers can write to a PR. We may want to fix the other part of this :) — in the meantime, can you attach a patch instead?

Reply:
You should be able to file a PR against this PR branch, no?

Reply:
@auscompgeek I've made those changes. On second thought, nobody push to this branch. The second command adds a large number of files, so I'm going to be forcing this branch to keep it as two separate, clean commits.